File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.*;
import java.util.*;

class Main {
    // 변수
    static int N, X;
    static int[] A;

    // 입력
    static void input() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        A = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        X = Integer.parseInt(br.readLine());

        br.close();
    }

    // 풀이
    static void pro() {
        // 배열 정렬
        Arrays.sort(A);

        int left = 0, right = N - 1, cnt = 0;

        while (left < right) {
            int sum = A[left] + A[right];

            if (sum < X) {
                left++;
                continue;
            }

            if (sum > X) {
                right--;
                continue;
            }

            if (sum == X) {
                cnt++;
                left++;
            }
        }

        // 정답 출력
        System.out.println(cnt);
    }

    // 실행
    public static void main(String[] args) throws Exception {
        input();
        pro();
    }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <iostream>
#include <algorithm>
using namespace std;

int N, X;
int A[100005];

void input() {
    cin >> N;

    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }
    sort(A, A + N);

    cin >> X;
}


void pro() {
    int left = 0, right = N - 1, cnt = 0;

    while (left < right) {
        int sum = A[left] + A[right];

        if (sum == X) {
            cnt++;
            left++;
            continue;
        }

        if (sum > X) {
            right--;
        } else {
            left++;
        }
    }

    cout << cnt;
}

int main() {
    // 최적화
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    input();
    pro();
}